objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,40 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)getAnyTypeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix], @"Invalid any type url.");
+  return [self.typeURL substringFromIndex: [GPBTypeGoogleApisComPrefix length]];
+}
+
+- (void)unpackTo:(GPBMessage*)message {
+  NSAssert([self is:[message class]],
+           [NSString stringWithFormat:@"%@/%@/%@/%@/%@",
+                    @"Given messsage's type name (",
+                    message.descriptor.name,
+                    ") doesn't match with any's (",
+                    self.getAnyTypeName,
+                    ")"]);
+  [message mergeFromData:self.value extensionRegistry:NULL];
+}
+
+- (void) packFrom:(GPBMessage*)message {
```

Comment:
If user care about the time to create an any instance and want to reuse any, packFrom will be more useful than initFromMessage. (packFrom will totally replace data in any)
In c++, we have a lot of such re-use. For example, in grpc, request/response are re-used instead of being created for each rpc call.
Does this also happen in objective-c?